digest.go

```diff
@@ -109,18 +109,11 @@ func (d Digest) Validate() error {
 		return ErrDigestInvalidFormat
```

Comment:
> This regexp check is no longer needed…

I think it is.  What if someone tries to make a digest with `foobar`?  Or `foobar:!!!`?  I'd rather get `ErrDigestInvalidFormat` for those than `ErrDigestUnsupported`.